### PR TITLE
Support for Frozen Sweep DPS and burst damage

### PR DESCRIPTION
**Pull request by Lilylicious**
### Description of the problem being solved:

* Adds frozen legion DPS
* Adds support for frozen legion enchants and alternate quality gems
* Fixes Sacrificial Zeal when mana cost is nil
* Adds burst damage to skill type-specific stats as well as the sidebar, currently works for Unleash and for Frozen Legion

### Link to a build that showcases this PR:
https://pobb.in/9o138R5jXhdy

### Before screenshot:
image

### After screenshot:
(with 75% chance to spawn additional statue)
image

image

image

### CODE REVIEW

**Comment by Lilylicious** [comment]
The main usecase for burst damage is for frozen legion right now. It's a skill where it's completely feasible to do 30m damage towards pinnacle bosses in one go. It's also a number that changes depending on the quality you have on your divergent gem, and with your helmet enchants, while average damage per hit doesn't change.

File: src/Export/Skills/act_str.txt [hunk 1]
```diff
@@ -632,10 +632,79 @@ local skills, mod, flag, skill = ...
...
+		
+		local statuesWaitedFor
+		local dpsMultiplier = ExtraHitChance
+
+		if activeSkill.skillPart == 1 then
+			statuesWaitedFor = 1
+		elseif activeSkill.skillPart == 2 then
+			statuesWaitedFor = MaxStatues
+		end
+		activeSkill.skillData.averageBurstHits = statuesWaitedFor + (ExtraHitChance > 0 and ExtraHitChance / 100 or 0)
```
Comment (QuickStick123) [inline src/Export/Skills/act_str.txt:694]:
```suggestion
		activeSkill.skillData.averageBurstHits = statuesWaitedFor + ExtraHitChance / 100
```
You have ensured this is always a number I don't see the reason to bother check it is greater than zero either.

File: src/Export/Skills/act_str.txt [hunk 2]
```diff
@@ -632,10 +632,79 @@ local skills, mod, flag, skill = ...
...
+		parentSkill.activeEffect.grantedEffect.castTime = 0
+		
+		activeSkill.skillData.cooldown = parentSkill.skillData.cooldown
+		local cooldownOverride = parentSkill.skillModList:Override(parentSkill.skillCfg, "CooldownRecovery")
+		local cooldown = cooldownOverride or (parentSkill.skillData.cooldown + parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "CooldownRecovery")) / m_max(0, calcLib.mod(parentSkill.skillModList, parentSkill.skillCfg, "CooldownRecovery"))
+		output.Cooldown = m_ceil(cooldown * data.misc.ServerTickRate) / data.misc.ServerTickRate
+		activeSkill.skillData.hitTimeOverride = output.Cooldown
+
+		local ExtraHitChance = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionExtraStatueChance") or 0
+		local MaxStatues = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionMaxStatues") or 1
```
Comment (QuickStick123) [inline src/Export/Skills/act_str.txt:684]:
```suggestion
		local extraStatueChance = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionExtraStatueChance") or 0
		local maxStatues = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionMaxStatues") or 1
```
We typically start with lower case for variables. I also like the internal mod name better. (it is also what you used in breakdown).

File: src/Modules/CalcOffence.lua [hunk 3]
```diff
@@ -2740,6 +2748,15 @@ function calcs.offence(env, actor, activeSkill)
 				t_insert(breakdown.AverageDamage, s_format("= %.1f", output.AverageDamage))
 			end
 		end
+		if globalBreakdown then
+			if globalOutput.AverageBurstDamage > 0 then
```
Comment (QuickStick123) [inline src/Modules/CalcOffence.lua:2752]:
```suggestion
		if globalBreakdown and globalOutput.AverageBurstDamage > 0 then
```
No need for two if statements.

**Review by LocalIdentity — APPROVED** [review]

**Pull request merged**